src/geometry/rotation_interpolation.rs

```diff
@@ -79,3 +82,67 @@ impl<T: SimdRealField> Rotation3<T> {
         q1.try_slerp(&q2, t, epsilon).map(|q| q.into())
     }
 }
+
+impl<T: RealField, const D: usize> Rotation<T, D>
+where
+    Const<D>: DimSub<U1>,
+    ArrayStorage<T, D, D>: Storage<T, Const<D>, Const<D>>,
+    DefaultAllocator: Allocator<T, Const<D>, Const<D>, Buffer = ArrayStorage<T, D, D>>
+        + Allocator<T, Const<D>>
+        + Allocator<T, Const<D>, DimDiff<Const<D>, U1>>
+        + Allocator<T, DimDiff<Const<D>, U1>>,
+{
+    ///
+    /// Computes the spherical linear interpolation between two general rotations.
+    ///
+    /// # Examples:
+    ///
+    /// ```
+    /// # use nalgebra::geometry::Rotation3;
+    ///
+    /// let q1 = Rotation3::from_euler_angles(std::f32::consts::FRAC_PI_4, 0.0, 0.0);
+    /// let q2 = Rotation3::from_euler_angles(-std::f32::consts::PI, 0.0, 0.0);
+    ///
+    /// let q = q1.slerp(&q2, 1.0 / 3.0);
+    ///
+    /// assert_eq!(q.euler_angles(), (std::f32::consts::FRAC_PI_2, 0.0, 0.0));
+    /// ```
+    ///
+    //FIXME: merging slerp for Rotation2 and Rotation3 into this raises the trait bounds
+    //from SimdRealField to RealField
+    #[inline]
+    #[must_use]
+    pub fn slerp(&self, other: &Self, t: T) -> Self {
+        use std::mem::transmute;
+
+        //The best option here would be to use #[feature(specialization)], but until
+        //that's stabilized, this is the best we can do. Theoretically, the compiler should
+        //pretty thoroughly optimize away all the excess checks and conversions
+        match D {
+            0 => self.clone(),
+
+            //FIXME: this doesn't really work in 1D since we can't interp between -1 and 1
+            1 => self.clone(),
+
+            //NOTE: Not pretty, but without refactoring the API, this is the best we can do
+            //NOTE: This is safe because we directly check the dimension first
+            2 => unsafe {
+                let (self2d, other2d) = (
+                    transmute::<&Self, &Rotation2<T>>(self),
+                    transmute::<&Self, &Rotation2<T>>(other),
```

Comment:
Is this sound? It seems like a reasonable operation in theory, just imagining the references as pointers-to-chunks-of-data, but I'm not confident enough in Rust's type aliasing rules to know if this violates some sort of compiler invariant.

Reply:
Yes, this is sound. Because this branch can only be dynamically reached when `Self` is `Rotaiton2<T>`, this is really just an identity operation. Leaving the lifetime unspecified is sketchy (this is probably transmuting to `&'static` implicitly), but still strictly sound since no dynamic access outlives the underlying borrow.

For easier auditing and more obvious correctness, extract the transmute into a safe helper function like:

```rs
fn identity_cast<T, D1, D2>(x: &Rotation<T, D1>) -> &Rotation<T, D2> {
    assert_eq!(D1, D2);
    mem::transmute(x)
}
```

Reply:
Well, I was _about_ to say that I tweaked the function to use `fixed_resize()` instead in order to remove the `transmute()`s, but I see what you're saying @Ralith

I'm good with either variant honestly. If we want to keep the code safe, I feel like the compiler _should_ be able to inline all of the copies. Otherwise, I'm down to revert back to the transmutes and add this signposting function if we want the performance guaranteed.

Reply:
I think a purely safe version is preferable unless a specific performance problem has been identified.